lib/topological_inventory/orchestrator/object_manager.rb

```diff
@@ -98,6 +106,39 @@ def manager_uri(path)
         )
       end
 
+      def check_deployment_config_quota(definition)
+        quota_status = kube_connection.get_resource_quota("compute-resources-non-terminating", my_namespace).status
+
+        cpu_limit = cpu_string_to_millicores(quota_status.used["limits.cpu"])
+        definition.dig(:spec, :template, :spec, :containers).each do |container|
+          cpu_limit += cpu_string_to_millicores(container.dig(:resources, :limits, :cpu))
+        end
+        raise(QuotaCpuLimitExceeded) if cpu_limit >= cpu_string_to_millicores(quota_status.hard["limits.cpu"])
+
+        cpu_request = cpu_string_to_millicores(quota_status.used["requests.cpu"])
+        definition.dig(:spec, :template, :spec, :containers).each do |container|
+          cpu_request += cpu_string_to_millicores(container.dig(:resources, :requests, :cpu))
+        end
+        raise(QuotaCpuRequestExceeded) if cpu_request >= cpu_string_to_millicores(quota_status.hard["requests.cpu"])
+
+        memory_limit = quota_status.used["limits.memory"].iec_60027_2_to_i
+        definition.dig(:spec, :template, :spec, :containers).each do |container|
+          memory_limit += container.dig(:resources, :limits, :memory).iec_60027_2_to_i
+        end
+        raise(QuotaMemoryLimitExceeded) if memory_limit >= quota_status.hard["limits.memory"].iec_60027_2_to_i
+
+        memory_request = quota_status.used["requests.memory"].iec_60027_2_to_i
+        definition.dig(:spec, :template, :spec, :containers).each do |container|
+          memory_request += container.dig(:resources, :requests, :memory).iec_60027_2_to_i
+        end
+        raise(QuotaMemoryRequestExceeded) if memory_request >= quota_status.hard["requests.memory"].iec_60027_2_to_i
+      end
+
+      def cpu_string_to_millicores(input)
+        match = input.match(/(?<number>\d+)(?<suffix>m?)/)
+        match[:suffix].empty? ? (match[:number].to_i * 1000) : match[:number].to_i
```

Comment:
Isn't this logic reversed? If we don't match a suffix then we have cores, right?

Reply:
Nope